lib/services/services.c

```diff
@@ -351,6 +419,12 @@ services__create_resource_action(const char *name, const char *standard,
                 }
                 op->opaque->args[index++] = crm_strdup_printf("--%s", key);
                 op->opaque->args[index++] = strdup(value);
+                if (op->opaque->args[index - 1] == NULL) {
```

Comment:
shouldn't we check both entries here?

In general looks if functionalizing something like prepare_resource_arguments would make sense.
Could return the number of arguments and caller checks them for existence or otherwise sets the ENOMEM stuff incl. crm_crit. Other error conditions set directly and return -1 upon other ENOMEM that doesn't reflect in arguments returns 0.

Reply:
crm_strdup_printf has an assertion ...

Reply:
yep :)

Reply:
It could definitely use more refactoring. I was thinking of one per standard (which would become an object method in that eventual redesign).